src/test/coins_tests.cpp

```diff
@@ -36,7 +37,7 @@ class CCoinsViewTest : public CCoinsView
     std::map<COutPoint, Coin> map_;
 
 public:
-    bool GetCoin(const COutPoint& outpoint, Coin& coin) const override
+    NODISCARD bool GetCoin(const COutPoint& outpoint, Coin& coin) const override
```

Comment:
Sorry, but this seems like a random change, why is this necessary?

Reply:
I'll let @MarcoFalke answer but personally I think it makes perfect sense to add `NODISCARD` to functions that return a `bool` indicating success/failure.

Reply:
There are tons of that?

Reply:
@promag Let me re-phrase: I think we should add `NODISCARD` to functions – especially new functions – where discarding the return value is likely a mistake. Think: `bool ParseFoo(const std::string& s, Foo *out);`). Sounds reasonable? :-)

Reply:
@practicalswift I'm not agains that annotation, I just don't understand why it's added to `GetCoin`.